protocols/messages.go

```diff
@@ -201,3 +202,50 @@ func (m Message) Summarize() MessageSummary {
 	}
 	return s
 }
+
+// These types are used to implement the zerolog.LogObjectMarshaler interface
+type RejectedIds []string
+type ObjectivePayloadSummaries []ObjectivePayloadSummary
+type ProposalSummaries []ProposalSummary
+type PaymentSummaries []PaymentSummary
+
+func (m MessageSummary) MarshalZerologObject(e *zerolog.Event) {
+	e.Str("To", m.To).
+		Array("PayloadSummaries", ObjectivePayloadSummaries(m.PayloadSummaries)).
+		Array("ProposalSummaries", ProposalSummaries(m.ProposalSummaries)).
+		Array("Payments", PaymentSummaries(m.Payments)).
+		Array("RejectedObjectives", RejectedIds(m.RejectedObjectives))
```

Comment:
I am probably missing something, but are these type casts needed?

Reply:
You're right, they're not needed! I think they were added by autocomplete. Removed in 182bcd8.